src/Native/Unix/System.Native/pal_signal.c

```diff
@@ -154,6 +141,10 @@ static void* SignalHandlerLoop(void* arg)
                 callback(reapAll ? 1 : 0);
             }
         }
+        else if (signalCode == SIGCONT)
+        {
+            ReinitializeTerminal();
+        }
         else
         {
             assert_msg(false, "invalid signalCode", (int)signalCode);
```

Comment:
Since you removed the filtering of which signals to send to the signal handling thread, isn't this assert now a problem?

Reply:
We have an `if` for each signal we are registering a handler for, so we shouldn't land in the `else` case.